clodius/tiles/bigbed.py

```diff
@@ -0,0 +1,239 @@
+
+from concurrent.futures import ThreadPoolExecutor
+
+DEFAULT_RANGE_MODE = 'significant'
+MIN_ELEMENTS = 1
+MAX_ELEMENTS = 50
+
+logger = logging.getLogger(__name__)
+
+range_modes = {}
+range_modes['significant'] = {'name': 'Significant', 'value': 'significant'}
+
+
+def tileset_info(bbpath, chromsizes=None):
+    ti = hgbw.tileset_info(bbpath, chromsizes)
+    ti['range_modes'] = range_modes
+    return ti
+
+
+def fetch_data(a):
+    (
+        bbpath,
+        binsize,
+        chromsizes,
+        range_mode,
+        min_elements,
+        max_elements,
+        cid,
+        start,
+        end
+    ) = a
+
+    try:
+        chrom = chromsizes.index[cid]
+
+        fetch_factory = ft.partial(bbi.fetch_intervals, bbpath, chrom, start, end)
+
+        if range_mode == 'significant':
+            intervals, intervals2 = fetch_factory(), fetch_factory()
+        else:
+            intervals, intervals2 = fetch_factory(), fetch_factory()
+
+    except IndexError:
+        # beyond the range of the available chromosomes
+        # probably means we've requested a range of absolute
+        # coordinates that stretch beyond the end of the genome
+        intervals, intervals2 = None, None
+
+    except KeyError:
+        # probably requested a chromosome that doesn't exist (e.g. chrM)
+        intervals, intervals2 = None, None
+
+    offset = 0
+    offsetIdx = 0
+    chrOffsets = {}
+    for chrSize in chromsizes:
+        chrOffsets[chromsizes.index[offsetIdx]] = offset
+        offset += chrSize
+        offsetIdx += 1
+
+    final_intervals = []
+    intervals_length = 0
+    scores = []
+
+    if not intervals:
+        return final_intervals
+
+    for interval in intervals:
+        scores.append(int(interval[4]))
+        intervals_length += 1
+
+    # generate beddb-like elements for parsing by the higlass plugin
+    if intervals_length >= min_elements and intervals_length <= max_elements:
+        for interval in intervals2:
+            final_intervals.append({'chrOffset': chrOffsets[chrom], 'importance': int(interval[4]), 'fields': interval})
+
+    elif intervals_length > max_elements:
+        thresholded_intervals = []
+        desired_perc = max_elements / intervals_length
+        thresholded_score = int(np.quantile(scores, 1 - desired_perc))
+        thresholded_intervals = [{'chrOffset': chrOffsets[chrom], 'importance': int(interval[4]), 'fields': interval} for interval in intervals2 if int(interval[4]) >= thresholded_score]
+        thresholded_intervals_length = len(thresholded_intervals)
+        if thresholded_intervals_length > max_elements:
+            indices = random.sample(range(thresholded_intervals_length), max_elements)
+            final_intervals = [thresholded_intervals[i] for i in sorted(indices)]
+
+    return final_intervals
+
+
+def get_bigbed_tile(
+    bbpath,
+    zoom_level,
+    start_pos,
+    end_pos,
+    chromsizes=None,
+    range_mode=None,
+    min_elements=None,
+    max_elements=None
+):
```

Comment:
I know we don't have a strong tradition of doing this right now, but perhaps you could add a docstring here describing what the return type looks like?